src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/JsonDictionaryConverter.cs

```diff
@@ -252,16 +252,17 @@ TKey ReadDictionaryKey(ref Utf8JsonReader reader, ref ReadStack state)
                 TKey key;
                 string unescapedPropertyNameAsString;
 
+                _keyConverter ??= GetConverter<TKey>(state.Current.JsonTypeInfo.KeyTypeInfo!);
```

Comment:
This could have been a previous optimization to avoid calling GetConverter().

Reply:
Indeed. I'm consoled that the fact that this need only happen once per dictionary converter instance so we should be ok.